HelloFlask/views.py

```diff
@@ -1,31 +1,44 @@
-from flask import Flask
-from flask import render_template
+from datetime import datetime
+from re import match
+
+from flask import Flask, render_template
+
 from HelloFlask import app
 
+
 @app.route('/')
 def home():
-    return render_template("home.html", title = "Home")
+    return render_template("home.html", title="Home")
 
 @app.route('/about')
 def about():
-    return render_template("about.html", title = "About us")
+    return render_template("about.html", title="About us")
 
 @app.route('/contact')
 def contact():
-    return render_template("contact.html", title = "Contact us")
+    return render_template("contact.html", title="Contact us")
 
 @app.route('/hello/<name>')
 def hello_there(name):
-    from datetime import datetime
-    now = datetime.now()
-
     return render_template(
         "hello_there.html",
         title ='Hello, Flask',
-        message = "Hello there, " + name + "!",
-        date = now.strftime("%A, %d %B, %Y at %X")
+        name = clean_name(name),
+        date = datetime.now()
     )
 
 @app.route('/api/data')
 def get_data():
-    return app.send_static_file('data.json')
\ No newline at end of file
+    return app.send_static_file('data.json')
+
+def clean_name(name):
+    # Filter the name argument to letters only using regular expressions. URL arguments
+    # can contain arbitrary text, so we restrict to safe characters only.
+    match_object = match("[a-zA-Z]+", name)
```

Comment:
If you decide to keep it, I would replace this whole function with the much more simple:

    return re.sub(r'[^a-zA-Z]+', '', name) or 'Friend'

the `r` prefix for the string is not needed here, but I think it's good practice to always use raw strings for regexps (so you don't forget it when adding something with a backslash later)

Reply:
I'll be omitting all that extra code anyway.